hax/hax/util.py

```diff
@@ -446,9 +446,54 @@ def get_service_keys(node_items: List[Any], fidk: int) -> List[Any]:
             if f'{fid}' == x['Key'].split('/')[-1]
         ]
 
+    def is_node_alive(self, node: str) -> bool:
+        """
+        Checks via Consul Members API whether the given node is alive.
+        """
+        try:
+            # Returns data of the following kind:
+            # [{
+            #     'Name': 'localhost',
+            #     'Addr': '192.168.6.214',
+            #     'Port': 8301,
+            #     'Tags': {
+            #         'acls': '0',
+            #         'bootstrap': '1',
+            #         'build': '1.7.8:9a5a1218',
+            #         'dc': 'dc1',
+            #         'id': 'dd8a91f6-ca32-30e0-983c-8f309d653045',
+            #         'port': '8300',
+            #         'raft_vsn': '3',
+            #         'role': 'consul',
+            #         'segment': '',
+            #         'vsn': '2',
+            #         'vsn_max': '3',
+            #         'vsn_min': '2',
+            #         'wan_join_port': '8302'
+            #     },
+            #     'Status': 1,
+            #     'ProtocolMin': 1,
+            #     'ProtocolMax': 5,
+            #     'ProtocolCur': 2,
+            #     'DelegateMin': 2,
+            #     'DelegateMax': 5,
+            #     'DelegateCur': 4
+            # }]
+            members_data = self.cns.agent.members()
+            LOG.log(TRACE, "members: %s", members_data)
+            for member in members_data:
+                if member['Name'] == node:
+                    return int(member['Status']) == 1
+            return True
```

Comment:
@mssawant Don't we need to return False here?

Reply:
Yes, sorry I changed the initial implementation and this was left out.

Reply:
@knekrasov, as per our today's discussion if are good to land this patch then can you please make this change and update it so that we can go ahead and merge it?

Reply:
Yes, I think so